lonestar/analytics/cpu/cdlp/cdlp_cli.cpp

```diff
@@ -0,0 +1,140 @@
+ */
+
+#include <iostream>
+
+#include "Lonestar/BoilerPlate.h"
+#include "katana/Mem.h"
+#include "katana/analytics/cdlp/cdlp.h"
+
+using namespace katana::analytics;
+
+namespace cll = llvm::cl;
+
+/// Limited number of iterations to limit the oscillation of the label
+/// in Synchronous algorithm. We dont need to limit it in Asynchronous algorithm.
+/// Set to 10 same as Graphalytics benchmark.
+/// FIXME: duplicated from cdlp.cpp; needs to defnied in one place. maybe in cdlp.hpp.
+const unsigned int kMaxIterations = 10;
+
+const char* name = "CDLP";
+const char* desc = "Detects the communities of a graph using label propagation";
+static const char* url = "cdlp";
+
+static cll::opt<std::string> inputFile(
+    cll::Positional, cll::desc("<input file>"), cll::Required);
+
+static cll::opt<size_t> maxIterations(
+    "maxIterations", cll::desc("Maximum number of running iterations"),
+    cll::init(kMaxIterations));
+
+static cll::opt<CdlpPlan::Algorithm> algo(
+    "algo", cll::desc("Choose an algorithm (default value Synchronous):"),
+    cll::values(
+        clEnumValN(
+            CdlpPlan::kSynchronous, "Synchronous",
+            "Synchronous algorithm")/*,
+
+		/// TODO: Asynchronous Algorithm will be implemented later after Synchronous
+		/// is done for both shared and distributed versions.
+		
+        clEnumValN(
+            CdlpPlan::kAsynchronous, "Asynchronous",
+            "Asynchronous algorithm")*/),
+    cll::init(CdlpPlan::kSynchronous));
+
+std::string
+AlgorithmName(CdlpPlan::Algorithm algorithm) {
+  switch (algorithm) {
+  case CdlpPlan::kSynchronous:
+    return "Synchronous";
+  /// TODO: Asynchronous Algorithm will be implemented later after Synchronous
+  /// is done for both shared and distributed versions.
+  /*
+  case CdlpPlan::kAsynchronous:
+    return "Asynchronous";
+  */
+  default:
+    return "Unknown";
+  }
+}
+
+int
+main(int argc, char** argv) {
+  std::unique_ptr<katana::SharedMemSys> G =
+      LonestarStart(argc, argv, name, desc, url, &inputFile);
+
+  katana::StatTimer totalTime("TimerTotal");
+  totalTime.start();
+
+  std::cout << "Reading from file: " << inputFile << "\n";
+  std::unique_ptr<katana::PropertyGraph> pg =
+      MakeFileGraph(inputFile, edge_property_name);
+
+  std::cout << "Read " << pg->topology().num_nodes() << " nodes, "
+            << pg->topology().num_edges() << " edges\n";
+
+  std::cout << "Running " << AlgorithmName(algo) << " algorithm\n";
+
+  CdlpPlan plan = CdlpPlan();
+  switch (algo) {
+  case CdlpPlan::kSynchronous:
+    plan = CdlpPlan::Synchronous();
+    break;
+  /// TODO: Asynchronous Algorithm will be implemented later after Synchronous
+  /// is done for both shared and distributed versions.
+  /*
+  case CdlpPlan::kAsynchronous:
+    plan = CdlpPlan::Asynchronous();
+    break;
+  */
+  default:
+    std::cerr << "Invalid algorithm\n";
+    abort();
+  }
+
+  auto pg_result = Cdlp(pg.get(), "community", maxIterations, plan);
```

Comment:
Use variable for output_property_name instead of a string "community". See enterprise lonestar apps for example. We will update other apps too.

Reply:
a const std::string has been defined for property name.